### ISSUE TITLE
Геокодирование до дома, до osm границы

### ISSUE THREAD

**Issue by d1g** [kiselev-dv/gazetteer#42]
Геокодирование по паре сотен POI внутри дома может выглядеть как overkill но для неё есть самое прямое применение в [OpenLevelUp](https://framagit.org/PanierAvide/OpenLevelUp). Главное чтобы osmid в ответе приходил.

##### Параметры API
В API gazetteer нужно добавить какой-нибудь параметр по которому будет ограничена область геокодинга.

osm_id=n00000001
osm_id=w00000001
osm_id=r00000001

Очевидно, что для этого клиентам нужно знать osmid (что противоречит геокодингу без костылей)

Возможно упростить всё пользователям двумя параметрами:
restrict=admin_level + restrict_query=Москва
restrict=building + restrict_query=Афимолл Сити

##### Индексация
Самая простая реализация это добавить метки is_in_Афимолл_Сити всем объектам внутри здания.

Для админ границ не такая востребованная фича, может подождать.

PS. Если это ещё не делается, желательно возвращать osmid в ответах

**Comment by kiselev-dv** [comment]
Я в выгрузках и так матчу здание (если оно само было тегировано) и пои, правда, возможно, я слишком многовыкидываю подробностей когда записываю конечный резульат.

**Comment by d1g** [comment]
Nominatim:
* [вот так я пытал](http://www.openstreetmap.org/search?query=Centre%20Commercial%20Le%20Coudoulet%20Sephora#map=19/44.12082/4.83917) - вообще 0
* [и наоборот](http://www.openstreetmap.org/search?query=Sephora%20Centre%20Commercial%20Le%20Coudoulet#map=19/44.12082/4.83917) - вообще 0

osm.me:
* начнём что [Афимолл Сити](http://osm.me/#!/ru/q/%D0%90%D1%84%D0%B8%D0%BC%D0%BE%D0%BB%D0%BB%20%D0%A1%D0%B8%D1%82%D0%B8/) возвращает одну деревню; отсюда и речь веду что моллы - как популярные "места" городов можно рассматривать
* дальше холось бы часть названия магаза подписать или в начало или в конец, тут дело хозяйское как параметры добавлять или всё в одну строку

**Comment by kiselev-dv** [comment]
Погоди, для начала, давай разберемся с выгрузками, что ищет osm.me это отдельный разговор (Мне так и не удалось добиться от ElasticSearch всего того что я хочу по поиску и сортировке результатов, так что находит он далеко не всегда то что я ожидаю).

Дай осмовскую айдишку для этого мола пожалуйста.

**Comment by d1g** [comment]
http://www.openstreetmap.org/way/134516768

**Comment by kiselev-dv** [comment]
Хорошо, мне надо посмотреть что именно я выгружаю для мола и для поек внутри. Это займет некоторое время.

**Comment by kiselev-dv** [comment]
Так вот что я выгружаю для Афимола https://gist.github.com/kiselev-dv/a24c3e6e498d142d905db5ddfe3db966

**Comment by d1g** [comment]
вижу есть данные по Starbucks: http://www.openstreetmap.org/node/1225311629
* не пустой "poi_class" (насколько я понял)
* "poi_addr_match": "contains"

Я ожидаю его увидеть по "Старбакс АФИМОЛЛ" "АФИМОЛЛ Старбакс" "Старбакс в афимолле"

**Comment by kiselev-dv** [comment]
Точно не старбакс в афимолле, вообще над генерацией текста я еще подумаю как это лучше сделать, сам объект я добавил https://gist.github.com/kiselev-dv/67737f2a0ff1abc24b88bd7ee4ae2595

**Comment by kiselev-dv** [comment]
Так, поехали дальше, @d1g  как ты хотел это использовать?

**Comment by d1g** [comment]
@kiselev-dv, изначальный сценарий был "вывести все POI внутри <АФИМОЛЛ>".

Затем возникла идея по айдишнику POI (n1225311629) возвращать ID здания. Третий вариант (магаз-здание) я недавно предложил.

overpass плох тем, что 1. он выводит по bbox (не попадает в экран часть магазов) 2. он требует знать название магаза (в случае area)

**Comment by kiselev-dv** [comment]
Так, `POI =>здание` будет работать как только обновлю, точнее айдишку и сейчас думаю можно вытащить она в рефах должна быть.

Все пои внутри здания, если я записываю айди здания для пои в рефы, то тоже уже сейчас должно работать.

Что не работает и пока я думаю как лучше - это текст. Ну точнее я отвлекся маленько, может на выходных прикручу текст.
